hplsql/src/main/java/org/apache/hive/hplsql/Exec.java

```diff
@@ -1870,12 +1878,30 @@ private boolean execUserSql(HplsqlParser.Expr_func_paramsContext ctx, String nam
         sql.append(", ");
       }
     }
-    sql.append(")");
+    sql.append(", \"");
+    sql.append(getStoredProcedure(name.toUpperCase()));
```

Comment:
What will happen if `getStoredProcedure(name.toUpperCase()) `returns null (in case the function is not registered as a HPL/SQL function). I feel this scenario is not handled, please check once.

Reply:
I think this is handled in the beginning of the method, [here](https://github.com/apache/hive/blob/7cd3107a76d633ef5fae2ffb8ec16953ac968092/hplsql/src/main/java/org/apache/hive/hplsql/Exec.java#L1849). It ultimately calls [this method](https://github.com/apache/hive/blob/7cd3107a76d633ef5fae2ffb8ec16953ac968092/hplsql/src/main/java/org/apache/hive/hplsql/functions/HmsFunctionRegistry.java#L148), and checks that the function is present. Since they both are using the same HMS API, I think this shouldn't be an issue, but let me know if we still need to check something here. :)

Reply:
Can you test a scenario like in HPL/SQL procedure/function you can use any Hive built-in function and check whether it works or not. Because [here](https://github.com/apache/hive/pull/4330/files#diff-7f870ba54791ec47f97a12f0b100d31a26527e5434a304edd61cb63ef9fb4596L108) earlier they were checking like if _functionDefinition_ null then they treat it as a built-in function and execute. So just once check this scenario, if it works then its fine.

Reply:
From my understanding, if we don't use a procedure and use just a udf instead, it will take a different code path, i.e., it won't use the hplsql udf. But, I have added two tests to check that udfs work well in hplsql mode. Please review again. :)